src/main/java/duke/command/CommandAddDeadline.java

```diff
@@ -0,0 +1,46 @@
+package duke.command;
+
+import java.time.format.DateTimeParseException;
+import java.util.regex.Matcher;
+
+import task.TaskDeadline;
+import task.TaskList;
+
+/**
+ * Command to add a deadline
+ */
+public class CommandAddDeadline extends Command {
+    private final TaskList taskList;
+    private final Matcher input;
+
+    /**
+     * Constructor
+     *
+     * @param taskList Task list
+     * @param groups Matcher with info needed to create the class
+     */
+    public CommandAddDeadline(TaskList taskList, Matcher groups) {
```

Comment:
`groups` could be abstracted out, or arguments can be extracted out into an array before passing into the Command constructor.

Reply:
That's true!